packages/frontend/src/components/Explorer/ExploreTree/TableTree/TableTreeSections.tsx

```diff
@@ -34,11 +34,13 @@ const TableTreeSections: FC<Props> = ({
     }, [table.metrics]);
 
     const customMetrics = useMemo(() => {
-        return additionalMetrics.reduce<Record<string, AdditionalMetric>>(
-            (acc, item) => ({ ...acc, [getItemId(item)]: item }),
-            {},
-        );
-    }, [additionalMetrics]);
+        return additionalMetrics
+            .filter((metric) => metric.table === table.name)
```

Comment:
moved the filtering here because it was causing a bug where: `hasCustomMetrics` comes back `false` even when we have some defined.
Since the filtering was happening on a map of tables, 2/3 were necessarily false and it was triggering a re-render.